crates/uv/src/commands/project/mod.rs

```diff
@@ -81,70 +83,83 @@ pub(crate) fn find_environment(
     PythonEnvironment::from_root(workspace.venv(), cache)
 }
 
-/// Initialize a virtual environment for the current project.
-pub(crate) fn init_environment(
+/// Check if the given environment satisfies the project's requirements.
+pub(crate) fn interpreter_meets_requirements(
+    interpreter: &Interpreter,
+    requested_python: Option<&str>,
+    requires_python: Option<&VersionSpecifiers>,
+    cache: &Cache,
+) -> bool {
+    // `--python` has highest precedence, after that we check `requires_python` from
+    // `pyproject.toml`. If `--python` and `requires_python` are mutually incompatible,
+    // we'll fail at the build or at last the install step when we aren't able to install
+    // the editable wheel for the current project into the venv.
+    // TODO(konsti): Do we want to support a workspace python version requirement?
+    if let Some(python) = requested_python {
+        let request = ToolchainRequest::parse(python);
+        if request.satisfied(interpreter, cache) {
+            debug!("Interpreter meets the requested python {}", request);
+            return true;
+        }
+
+        debug!("Interpreter does not meet the request {}", request);
+        return false;
+    };
+
+    if let Some(requires_python) = requires_python {
+        if requires_python.contains(interpreter.python_version()) {
+            debug!(
+                "Interpreter meets the project `Requires-Python` constraint {}",
+                requires_python
+            );
+            return true;
+        }
+
+        debug!(
+            "Interpreter does not meet the project `Requires-Python` constraint {}",
+            requires_python
+        );
+        return false;
+    };
+
+    // No requirement to check
+    true
+}
+
+/// Find the interpreter to use in the current project.
+pub(crate) fn find_interpreter(
     workspace: &Workspace,
     python: Option<&str>,
-    preview: PreviewMode,
```

Comment:
We should assume preview is enabled during preview commands for now.

Reply:
(i.e. if I'm using the experimental interface I shouldn't need to opt-in to managed toolchains)